ddtrace/profiling/recorder.py

```diff
@@ -1,17 +1,38 @@
 # -*- encoding: utf-8 -*-
 import collections
+import threading
 
-from ddtrace.profiling import _attr
 from ddtrace.vendor import attr
 
 
+class _defaultdictkey(dict):
+    """A variant of defaultdict that calls default_factory with the missing key as argument."""
+
+    def __init__(self, default_factory=None):
+        self.default_factory = default_factory
+
+    def __missing__(self, key):
+        if self.default_factory:
+            v = self[key] = self.default_factory(key)
+            return v
+        raise KeyError(key)
+
+
 @attr.s(slots=True, eq=False)
 class Recorder(object):
     """An object that records program activity."""
 
+    _DEFAULT_MAX_SIZE = 32768
+
+    default_max_size = attr.ib(default=_DEFAULT_MAX_SIZE)
```

Comment:
Would `default_max_events` make more sense here? With the class being `Recorder` it's not well-defined what "size" means in my opinion. (Also we use `DD_PROFILING_MAX_EVENTS` already)

Reply:
Sure